src/codewhisperer/util/authUtil.ts

```diff
@@ -204,6 +204,10 @@ export class AuthUtil {
     }
 
     public isConnectionValid(): boolean {
+        if (!this.isConnected()) {
```

Comment:
I may have caused confusion in my slack comment, this wasn't what I meant. We should be checking if `isConnectionValid()` in the callback functions of the commands that hijack the keybindings